tools/golden_tests_harvester/lib/src/digests_json_format.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert' as convert;
+
+/// A Dart representation of a `digests.json` file.
+///
+/// A single file is typically used to represent the results of a test suite,
+/// or a series of tests that have been run with the same [dimensions]. For
+/// example, "Impeller Unittests on Vulkan running on MacOS" might generate
+/// a `digests.json` file.
+///
+/// Other tools (perhaps implemented in other languages, like C++) can use this
+/// format to communicate with the `golden_tests_harvester` tool without
+/// relying on the tool directly (or the Dart SDK).
+final class Digests {
```

Comment:
This code seems like the sort of thing we should be automating, can we use the dart json parser to do this?  We would probably need custom builders which is a showstopper, eh?

Reply:
I am guessing (?) this format will virtually never change, or if it does, we'll catch it in a unit test.

I am not opposed to generating this, but it wouldn't be a priority for me.